src/ngraph/descriptor/tensor_view.hpp

```diff
@@ -35,35 +35,22 @@ namespace ngraph
             class TensorViewLayout;
         }
 
-        class Tensor;
-        class TensorView;
-
         /// @brief Compile-time descriptor of a first-class value that is a view of a tensor.
         class TensorView
         {
             TensorView(const TensorView&) = delete;
             TensorView& operator=(const TensorView&) = delete;
 
-        protected:
-            TensorView(const std::shared_ptr<const TensorViewType>& tensor_view_type)
-                : m_tensor_view_type(tensor_view_type)
-            {
-            }
-
         public:
-            virtual ~TensorView() {}
-            virtual const Tensor& get_tensor() const = 0;
-            virtual Tensor& get_tensor() = 0;
+            TensorView(const element::Type& element_type,
+                       const Shape& shape,
+                       const std::string& name);
 
-            virtual std::shared_ptr<const TensorViewType> get_value_type() const;
-            const std::string& get_name() const { return m_name; }
-            std::shared_ptr<const TensorViewType> get_tensor_view_type() const
-            {
-                return m_tensor_view_type;
-            }
+            const Tensor& get_tensor() const;
+            Tensor& get_tensor();
 
-            virtual void set_tensor_view_type(const element::Type& element_type,
-                                              const Shape& shape);
+            const std::string& get_name() const { return m_name; }
+            void set_tensor_view_type(const element::Type& element_type, const Shape& shape);
```

Comment:
Should we break it up into `set_element_type` and `set_shape`?

Reply:
So far, I have never seen a case where one was set and the other was not.

Reply:
Sure, because before `set_tensor_view_type` was the only way.  ;)